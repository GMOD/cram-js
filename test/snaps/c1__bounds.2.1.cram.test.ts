import { testFile } from '../testUtil'
testFile('c1#bounds.2.1.cram')
