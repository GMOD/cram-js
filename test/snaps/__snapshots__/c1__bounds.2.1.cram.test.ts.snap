// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole c1#bounds.2.1.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 2,
    "minorVersion": 1,
  },
  {
    "containerHeader": {
      "_endPosition": 39,
      "_size": 13,
      "alignmentSpan": 0,
      "landmarks": [
        0,
      ],
      "length": 10098,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 1,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 39,
        "_size": 13,
        "alignmentSpan": 0,
        "landmarks": [
          0,
        ],
        "length": 10098,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 1,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "BA": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              67,
              71,
            ],
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              66,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              2,
              0,
              1,
            ],
          },
          "parametersBytes": 8,
        },
        "FP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              10,
              1,
              9,
            ],
          },
          "parametersBytes": 8,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 10153,
      "_size": 14,
      "alignmentSpan": 10,
      "landmarks": [
        202,
      ],
      "length": 325,
      "numBases": 30,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 3,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 10153,
        "_size": 14,
        "alignmentSpan": 10,
        "landmarks": [
          202,
        ],
        "length": 325,
        "numBases": 30,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 3,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10361,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "B",
                "data": [
                  "C",
                  9,
                ],
                "pos": 10,
                "refPos": 11,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10362,
          },
          {
            "alignmentStart": 3,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "B",
                "data": [
                  "C",
                  9,
                ],
                "pos": 9,
                "refPos": 11,
              },
              {
                "code": "B",
                "data": [
                  "G",
                  9,
                ],
                "pos": 10,
                "refPos": 12,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s2",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10363,
          },
        ],
        "header": {
          "_endPosition": 10390,
          "_size": 30,
          "compressedSize": 30,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            10,
            3,
            0,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            226,
            36,
            189,
            138,
            112,
            229,
            70,
            108,
            106,
            3,
            59,
            232,
            134,
            4,
            92,
            133,
          ],
          "contentId": 0,
          "contentPosition": 10360,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10183,
            "_size": 30,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              226,
              36,
              189,
              138,
              112,
              229,
              70,
              108,
              106,
              3,
              59,
              232,
              134,
              4,
              92,
              133,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 3,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 10,
            "refSeqStart": 1,
          },
          "uncompressedSize": 30,
        },
      },
    ],
  },
]
`;
